Add basic UI

zamcham opened the pull request:
In this PR I created the `app.rb` file that serves as the console app entry-point and `main.rb` that defines the entry point with a method called `main` that is invoked at the end of the file.

The method does the following:
- Present the user with a list of options to perform.
- Lets users choose an option.
- If needed, ask for parameters for the option.
- Have a way to quit the app.

zamcham committed:
Create app.rb

defines a class named App with several methods for managing people, books, and rentals. It uses the require_relative statements to import the necessary classes from separate files.

zamcham committed:
Create main.rb

ishimwezachee commented on app.rb:
```diff
@@ -0,0 +1,126 @@
+    if book.nil?
+      puts 'Book not found.'
+      return
+    end
+    print 'Enter the rental date (YYYY-MM-DD): '
+    date = gets.chomp
+    rental = Rental.new(book, person, date)
+    @rentals << rental
+    puts 'Rental created successfully.'
+  end
+
+  def list_rentals_for_person
+    puts 'Select a person to list rentals:'
+    list_all_people
+
+    print 'Enter the person\'s ID: '
+    person_id = gets.chomp.to_i
+
+    person = @people.find { |p| p.id == person_id }
+    if person.nil?
+      puts 'Person not found.'
+      return
+    end
+
+    rentals = @rentals.select { |r| r.person == person }
+
+    if rentals.empty?
+      puts 'No rentals found for the person.'
+    else
+      puts "Rentals for Person #{person.id} (#{person.name}):"
+      rentals.each do |rental|
+        puts "Book: #{rental.book.title}, Date: #{rental.date}"
+      end
+    end
+  end
+
+  def quit
+    false
+  end
+end
```
- [ ]  You did great to work on the UI but unfortunately I couldn't create a person. I could pass the name and the age, but both specialization and permission are causing the error. Fix the error and make sure that rental can be created successfully as it is required. 
![image](https://github.com/zamcham/OOP-school-library/assets/43877689/41932d37-e688-4f45-98a3-2bbae320f189)

ishimwezachee reviewed (changes requested):
Hi @zamcham, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
![image](https://media0.giphy.com/media/t8YPf3ACKeouLzEIvr/giphy.gif)

## Highlights
✅ book is created successfully 
✅  Project is running in the terminal 
✅ Linters are passing 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

zamcham committed:
Update teacher.rb parameters in initialize method

zamcham committed:
Update app.rb

the create_person method now passes the name and age as named arguments when creating a Teacher object. This ensures that the correct arguments are passed to the Teacher constructor.

jerryowusu commented on app.rb:
```diff
@@ -0,0 +1,126 @@
+  def create_person
+    print 'Do you want to create a Student(1) or a Teacher(2)? [Input the number]: '
+    res = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    if res == 1
+      print 'Has parent permission? [Y/N]: '
+      parent_permission = gets.chomp.downcase == 'y'
+      person = Student.new(age, nil, name, parent_permission: parent_permission)
+      puts "Student created successfully. ID: #{person.id}, Name: #{person.name}"
+    elsif res == 2
+      print 'Enter the teacher\'s specialization: '
+      specialization = gets.chomp
+      person = Teacher.new(nil, specialization, name: name, age: age)
+      puts "Teacher created successfully. Name: #{person.name}, ID: #{person.id}"
+    else
+      puts 'Invalid choice. Person creation failed.'
+    end
+    @people << person
+  end
+
+  def create_book
+    print 'Enter the book\'s title: '
+    title = gets.chomp
+    print 'Enter the book\'s author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully.'
+  end
+
+  def select_person
+    puts 'Select a person to create a rental:'
+    list_all_people
+    print 'Enter the person\'s ID: '
+    person_id = gets.chomp.to_i
+    @people.find { |p| p.id == person_id }
+  end
```
Your app shows an error when you try to create a person as shown in the image below. Kindly ensure that you give the correct number of arguments to the student class since that is the cause of the error.


![Screenshot from 2023-06-29 23-07-12](https://github.com/zamcham/OOP-school-library/assets/65233860/0a27e917-f3db-4863-9c57-0ccda65f8ac1)

jerryowusu reviewed (changes requested):
Hi @zamcham , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![almost-there-reach](https://github.com/zamcham/OOP-school-library/assets/65233860/dfcd98a2-9d27-45a8-80a2-2e7ab1fa6f64)

 ## Highlights
Book created  :+1: 
Nice design :+1: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

zamcham committed:
Update app.rb

zamcham committed:
Update book.rb

zamcham committed:
Update classroom.rb

remove space

zamcham committed:
refactor main.rb

zamcham committed:
Update person.rb

zamcham committed:
Update rental.rb

zamcham committed:
refactor student.rb

zamcham committed:
refactor teacher.rb

nahnahdev reviewed (approved):
## Project Status: APPROVED :green_circle: 
Hi @zamcham  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
<img src="https://media1.giphy.com/media/cL4pqu8GGRIihabgSM/giphy.gif"/>

## Highlights 🔥 
- School library console app behaves in the same way as the example video ✔️ 
- Linters well setup and passing ✔️
- Good use of GitFlow ✔️
- App entry point correctly setup ✔️
- All requested changes implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
